tests/ProductionFilterTest.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+namespace RonasIT\TelescopeExtension\Tests;
+
+use Laravel\Telescope\EntryType;
+use Laravel\Telescope\IncomingEntry;
+use Laravel\Telescope\Watchers\RequestWatcher;
+use RonasIT\TelescopeExtension\Filters\ProductionFilter;
+use RonasIT\TelescopeExtension\Tests\Support\ProductionFilterTestTrait;
+use Symfony\Component\HttpFoundation\Response;
+
+class ProductionFilterTest extends TestCase
+{
+    use ProductionFilterTestTrait;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        config(['telescope.watchers.' . RequestWatcher::class => []]);
+    }
+
+    public function testDevEnv()
+    {
+        $this->mockEnvironment('development');
+
+        $filter = new ProductionFilter();
+
+        $closure = $filter->apply();
+
+        $this->assertTrue($closure(new IncomingEntry([])));
+    }
+
+    public function testLocalEnv()
+    {
+        $this->mockEnvironment('local');
+
+        $filter = new ProductionFilter();
+
+        $closure = $filter->apply();
+
+        $this->assertTrue($closure(new IncomingEntry([])));
+    }
+
+    public function testExceptionProdEnv()
+    {
+        $this->mockEnvironment('production');
+
+        $filter = new ProductionFilter();
+
+        $entry = new IncomingEntry([]);
+        $entry->type(EntryType::EXCEPTION);
+
+        $closure = $filter->apply();
+
+        $this->assertTrue($closure($entry));
+    }
+
+    public function testSuccessRequestProdEnv()
+    {
+        $this->mockEnvironment('production');
+
+        $filter = new ProductionFilter();
+
+        $entry = new IncomingEntry([]);
+        $entry->type(EntryType::REQUEST);
```

Comment:
let's create a separate `IncomingRequest` class inside the `tests/Support/Mock` folder which will receive the `status` and `responseData` constructor arguments